File: app/src/main/java/com/example/installer/entity/PackageEntity.java
package com.example.installer.entity;


import java.text.SimpleDateFormat;
import java.util.Locale;

/**
 * Author:  ZhangTao
 * Date: 2018/3/5.
 */

public class PackageEntity implements ISelectable {

    /**
     * id : 12
     * application_name : 南京麦风
     * application_describe : null
     * icon_url : http://cdn.mse.mlwplus.com/meiliwu/applications/2017/11/30/09/50/Spotlight.png
     * bundle_id : com.eallcn.chowRentAgent
     * testing_bundle_id : com.eallcn.chowRentAgentDev
     * if_deleted : 0
     * create_time : 1512006618
     * update_time : 1512006618
     */
    private String id;
    private String application_name;
    private Object application_describe;
    private String icon_url;
    private String bundle_id;
    private String testing_bundle_id;
    private String if_deleted;
    private String create_time;
    private String update_time;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getApplication_name() {
        return application_name;
    }

    public void setApplication_name(String application_name) {
        this.application_name = application_name;
    }

    public Object getApplication_describe() {
        return application_describe;
    }

    public void setApplication_describe(Object application_describe) {
        this.application_describe = application_describe;
    }

    public String getIcon_url() {
        return icon_url;
    }

    public void setIcon_url(String icon_url) {
        this.icon_url = icon_url;
    }

    public String getBundle_id() {
        return bundle_id;
    }

    public void setBundle_id(String bundle_id) {
        this.bundle_id = bundle_id;
    }

    public String getTesting_bundle_id() {
        return testing_bundle_id;
    }

    public void setTesting_bundle_id(String testing_bundle_id) {
        this.testing_bundle_id = testing_bundle_id;
    }

    public String getIf_deleted() {
        return if_deleted;
    }

    public void setIf_deleted(String if_deleted) {
        this.if_deleted = if_deleted;
    }

    public String getCreate_time() {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss", Locale.getDefault());
        String format = simpleDateFormat.format(Long.valueOf(create_time) * 1000);
        return format;
    }

    public void setCreate_time(String create_time) {
        this.create_time = create_time;
    }

    public String getUpdate_time() {
        return update_time;
    }

    public void setUpdate_time(String update_time) {
        this.update_time = update_time;
    }

    @Override
    public String toString() {
        return "PackageEntity{" +
                "id='" + id + '\'' +
                ", application_name='" + application_name + '\'' +
                ", application_describe=" + application_describe +
                ", icon_url='" + icon_url + '\'' +
                ", bundle_id='" + bundle_id + '\'' +
                ", testing_bundle_id='" + testing_bundle_id + '\'' +
                ", if_deleted='" + if_deleted + '\'' +
                ", create_time='" + create_time + '\'' +
                ", update_time='" + update_time + '\'' +
                '}';
    }

    @Override
    public String getID() {
        return this.getId();
    }

    @Override
    public String getName() {
        return this.getApplication_name();
    }
}

File: app/src/main/java/com/example/installer/entity/PackageEntity.kt
package com.example.installer.entity

import java.text.SimpleDateFormat
import java.util.*

/**
@author zhangtao
@date   2020/10/23

 **/
data class PackageEntity(var id: String?) : ISelectable {
    /**
     * id : 464
     * version_name : 4.9.3
     * application_id : 1
     * application_name : 美丽屋经纪人
     * system_name : android
     * version_describe : null
     * download_url : http://cdn.mse.mlwplus.com/meiliwu/applications/2018/02/02/18/37/mlw_release_V4.9.3_2018_02_02_18.apk
     * plist_url :
     * version_type : 正式
     * if_deleted : 0
     * status : 1
     * create_time : 1517550821
     * update_time : 1517567888
     * update_type : 1
     * uid : 50200d0ba8207c4b80a17744
     * icon_url : http://cdn.mse.mlwplus.com/meiliwu/applications/2018/03/05/18/23/jingjiren.png
     */
    var version_name: String? = null
    var application_id: String? = null
    var application_name: String? = null
        get() {
            return field + "(" + version_name + ")"
        }
    var system_name: String? = null
    var download_url: String? = null
    var version_type: String? = null
    var create_time: String? = null
        get() {
            val simpleDateFormat = SimpleDateFormat("yyyy-MM-dd hh:mm:ss", Locale.getDefault())
            field?.let { return simpleDateFormat.format(field?.toLong()?.times(1000)) }
            return null
        }


    var icon_url: String? = null
    override fun getID(): String? {
        return id
    }

    override fun getName(): String? {
        return application_name
    }
}